flag_engine/features/models.py

```diff
@@ -113,18 +113,19 @@ def is_higher_segment_priority(self, other: "FeatureStateModel") -> bool:
 
         """
 
-        try:
-            return (
-                getattr(
-                    self.feature_segment,
-                    "priority",
-                    math.inf,
+        if other_feature_segment := other.feature_segment:
+            if (
+                other_feature_segment_priority := other_feature_segment.priority
+            ) is not None:
+                return (
+                    getattr(
+                        self.feature_segment,
+                        "priority",
+                        math.inf,
+                    )
+                    < other_feature_segment_priority
                 )
-                < other.feature_segment.priority
-            )
-
-        except (TypeError, AttributeError):
-            return False
```

Comment:
I understand these were intentionally removed, but why were they here in the first place?

Reply:
It was a guard against `other.feature_segment is None` or `other.feature_segment.priority is None`. Walrus magic does this now.